src/NUnitFramework/framework/Internal/Execution/SimpleWorkItemDispatcher.cs

```diff
@@ -60,7 +60,16 @@ public void Start(WorkItem topLevelWorkItem)
 
 #if APARTMENT_STATE
             if (topLevelWorkItem.TargetApartment == ApartmentState.STA)
+            {
+#if NETSTANDARD2_0
+                if (!OSPlatform.CurrentPlatform.IsWindows)
```

Comment:
This is a lot more expensive than `RuntimeInformation.IsOSPlatform(OSPlatform.Windows)` (needs `using OSPlatform = System.Runtime.InteropServices.OSPlatform`) or catching PlatformNotSupportedException.